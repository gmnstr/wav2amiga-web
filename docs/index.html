<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Wav2Amiga - Convert WAV to Amiga 8SVX</title>
    <style>
        body {
            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, sans-serif;
            line-height: 1.6;
            margin: 0;
            padding: 0;
            background: #f8f9fa;
            color: #333;
        }
        .container {
            max-width: 1200px;
            margin: 0 auto;
            padding: 2rem;
        }
        .header {
            text-align: center;
            margin-bottom: 3rem;
            padding: 2rem 0;
            background: white;
            border-radius: 8px;
            box-shadow: 0 2px 4px rgba(0,0,0,0.1);
        }
        .header h1 {
            color: #2c3e50;
            margin: 0 0 1rem 0;
            font-size: 2.5rem;
        }
        .header p {
            font-size: 1.2rem;
            color: #666;
            margin: 0;
        }
        .badges {
            margin: 1rem 0;
        }
        .badges a {
            margin: 0 0.5rem;
            text-decoration: none;
        }
        .cta {
            display: flex;
            gap: 1rem;
            justify-content: center;
            margin: 2rem 0;
        }
        .btn {
            display: inline-block;
            padding: 1rem 2rem;
            background: #3498db;
            color: white;
            text-decoration: none;
            border-radius: 6px;
            font-weight: 600;
            transition: background 0.3s;
        }
        .btn:hover {
            background: #2980b9;
        }
        .btn.secondary {
            background: #95a5a6;
        }
        .btn.secondary:hover {
            background: #7f8c8d;
        }
        .features {
            display: grid;
            grid-template-columns: repeat(auto-fit, minmax(300px, 1fr));
            gap: 2rem;
            margin: 3rem 0;
        }
        .feature {
            background: white;
            padding: 2rem;
            border-radius: 8px;
            box-shadow: 0 2px 4px rgba(0,0,0,0.1);
        }
        .feature h3 {
            color: #2c3e50;
            margin-top: 0;
        }
        .code {
            background: #f4f4f4;
            padding: 1rem;
            border-radius: 4px;
            font-family: 'Monaco', 'Menlo', monospace;
            overflow-x: auto;
            margin: 1rem 0;
        }
        .footer {
            text-align: center;
            margin-top: 3rem;
            padding: 2rem;
            color: #666;
            border-top: 1px solid #eee;
        }
    </style>
</head>
<body>
    <div class="container">
        <div class="header">
            <h1>Wav2Amiga</h1>
            <p>Convert WAV files to Amiga 8SVX format with high-quality resampling</p>
            <div class="badges">
                <a href="https://github.com/gmnstr/wav2amiga-web/actions/workflows/ci.yml">
                    <img src="https://github.com/gmnstr/wav2amiga-web/actions/workflows/ci.yml/badge.svg?branch=main" alt="CI">
                </a>
                <a href="https://github.com/gmnstr/wav2amiga-web">
                    <img src="https://img.shields.io/badge/version-v0.1.0-blue" alt="Version">
                </a>
            </div>
            <div class="cta">
                <a href="./app/" class="btn">Try Web Converter</a>
                <a href="https://github.com/gmnstr/wav2amiga-web" class="btn secondary">View on GitHub</a>
            </div>
        </div>

        <div class="features">
            <div class="feature">
                <h3>üéµ ZOH Resampling</h3>
                <p>Zero-order hold resampling preserves transients without interpolation or low-pass filtering, matching Paula chip sample-and-hold behavior.</p>
            </div>
            <div class="feature">
                <h3>üåê Cross-Platform</h3>
                <p>CLI tool for batch conversion and web interface for browser-based conversion with drag-and-drop support.</p>
            </div>
            <div class="feature">
                <h3>üéØ Deterministic</h3>
                <p>Byte-identical output across all platforms through pure integer math and pinned toolchain versions.</p>
            </div>
            <div class="feature">
                <h3>üì¶ Multiple Modes</h3>
                <p>Single, stacked, and stacked-equal sample layouts for different ProTracker use cases.</p>
            </div>
        </div>

        <div class="feature">
            <h3>Quick Start</h3>
            <p>Clone and build from source:</p>
            <div class="code">git clone https://github.com/gmnstr/wav2amiga-web.git
cd wav2amiga-web
pnpm install
pnpm build</div>
            <p>Convert a single sample:</p>
            <div class="code">node apps/cli/dist/cli.js --mode single --note C-2 input.wav</div>
            <p>Or use the <a href="./app/">web interface</a> for drag-and-drop conversion.</p>
        </div>

        <div class="footer">
            <p>For complete documentation, see <a href="https://github.com/gmnstr/wav2amiga-web/tree/main/docs">docs/</a></p>
            <p>MIT License ‚Ä¢ <a href="https://github.com/gmnstr/wav2amiga-web">GitHub</a></p>
        </div>
    </div>
</body>
</html>
