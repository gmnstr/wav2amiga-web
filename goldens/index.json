{
  "version": 2,
  "toolchain": {
    "node": "v20.17.0",
    "pnpm": "9.10.0",
    "ffmpeg": "ffmpeg version 6.0 Copyright (c) 2000-2023 the FFmpeg developers\nbuilt with Apple clang version 13.1.6 (clang-1316.0.21.2.5)\nconfiguration: --prefix=/Volumes/tempdisk/sw --extra-cflags=-fno-stack-check --arch=arm64 --cc=/usr/bin/clang --enable-gpl --enable-libvmaf --enable-libbluray --enable-libopenjpeg --enable-libopus --enable-libmp3lame --enable-libx264 --enable-libx265 --enable-libvpx --enable-libwebp --enable-libass --enable-libfreetype --enable-fontconfig --enable-libtheora --enable-libvorbis --enable-libsnappy --enable-libaom --enable-libvidstab --enable-libzimg --enable-libsvtav1 --enable-libkvazaar --enable-version3 --pkg-config-flags=--static --enable-ffplay --enable-postproc --enable-nonfree --enable-neon --enable-runtime-cpudetect --disable-indev=qtkit --disable-indev=x11grab_xcb\nlibavutil      58.  2.100 / 58.  2.100\nlibavcodec     60.  3.100 / 60.  3.100\nlibavformat    60.  3.100 / 60.  3.100\nlibavdevice    60.  1.100 / 60.  1.100\nlibavfilter     9.  3.100 /  9.  3.100\nlibswscale      7.  1.100 /  7.  1.100\nlibswresample   4. 10.100 /  4. 10.100\nlibpostproc    57.  1.100 / 57.  1.100",
    "resampler": {
      "name": "zoh",
      "version": "1.0.0"
    },
    "git": "a723cd80a43bfd06e2af07c668c7f7a915fee03f"
  },
  "cases": [
    {
      "id": "single_c2_sine",
      "mode": "single",
      "note": "C-2",
      "inputs": [
        {
          "path": "cases/single_c2_sine/input.wav",
          "sha256": "939ff6ec35e1e5e54f565772efa630d165a2eda368abdbef020272ba4179bb5d"
        }
      ],
      "expect": {
        "output": {
          "path": "cases/single_c2_sine/output.8SVX",
          "sha256": "a19dc46d2a4399f0ac48901a3dfe8e93dc160b780668526738d0eaba80af7add"
        },
        "report": {
          "path": "cases/single_c2_sine/output_report.json",
          "sha256": "c7b8fa8b9d2a8a0f9777586a5a038d7befb9316629c6d2f2432bc512a1d33bcc"
        }
      }
    },
    {
      "id": "stacked_offsets",
      "mode": "stacked",
      "manifest": "cases/stacked_offsets/manifest.json",
      "inputs": [
        {
          "path": "cases/stacked_offsets/input_a.wav",
          "sha256": "40b1a6a0f51a59bd43cf8f04ef4037fc552094058ad6865d9ade5629d3b2a289"
        },
        {
          "path": "cases/stacked_offsets/input_b.wav",
          "sha256": "a0212ca6c8f754072c98e2b0b05d01d42cf7dc2dad5da32be29d9abb2c419271"
        },
        {
          "path": "cases/stacked_offsets/input_c.wav",
          "sha256": "4e4b51ef630d3a6c989a257eaea581d20a1e11300ae73354557a9bc217cb85d2"
        }
      ],
      "expect": {
        "output": {
          "path": "cases/stacked_offsets/output_00_05_09.8SVX",
          "sha256": "ab1407b4a05d1b8478ad29c30cd4dc13921bb66dcdd82bcc6f1ddedc27587dfb"
        },
        "report": {
          "path": "cases/stacked_offsets/output_00_05_09_report.json",
          "sha256": "ce763b27ce2d64daf74686c7ef7e58e8e1d468f999a81a0e78d709e8d79d75e0"
        }
      }
    },
    {
      "id": "stackedequal_uniform",
      "mode": "stacked-equal",
      "manifest": "cases/stackedequal_uniform/manifest.json",
      "inputs": [
        {
          "path": "cases/stackedequal_uniform/input_a.wav",
          "sha256": "28645d7b0b4d9cb50fce9dd9b7ad9b20fbc8077aae79635964824de7334bfeb3"
        },
        {
          "path": "cases/stackedequal_uniform/input_b.wav",
          "sha256": "ef7832de4f3fcbfa44bdacb70e4364f925684d5c5ef02f465e09e256936cb8cb"
        },
        {
          "path": "cases/stackedequal_uniform/input_c.wav",
          "sha256": "bef157bd326762b4cd443d5a1ca38e3a8cedcaa2f526610b61c34022657c1c2f"
        }
      ],
      "expect": {
        "output": {
          "path": "cases/stackedequal_uniform/output_05.8SVX",
          "sha256": "e52ac7d2832f72988cbf6a0dbcffe9ad9d99aacd5f7189958bd214145a9fd8a1"
        },
        "report": {
          "path": "cases/stackedequal_uniform/output_05_report.json",
          "sha256": "a4c864e9a6699b028f6e08bff602270d659485cad4f31a7722e9225814bf382e"
        }
      }
    },
    {
      "id": "aligned_boundary",
      "mode": "single",
      "note": "C-2",
      "inputs": [
        {
          "path": "cases/aligned_boundary/input.wav",
          "sha256": "b735f4f94f483526a389d6f46b776e7c0f36be0c691e89ef0292480405d0dbc4"
        }
      ],
      "expect": {
        "output": {
          "path": "cases/aligned_boundary/output.8SVX",
          "sha256": "fdf53622292eab662f6a30ae977d31dc2c59d63794b29d0fbaf337b42c500075"
        },
        "report": {
          "path": "cases/aligned_boundary/output_report.json",
          "sha256": "9dfec59f545b73d6b73132387e90109ad0546bf6b1f6b2097bf866cdb5b2c39a"
        }
      }
    },
    {
      "id": "oversize_warning",
      "mode": "single",
      "note": "C-2",
      "inputs": [
        {
          "path": "cases/oversize_warning/input.wav",
          "sha256": "655324ecd4cfdf3ec6deb321e770b009664a286ba046624a3b0b96cb73c21b61"
        }
      ],
      "expect": {
        "output": {
          "path": "cases/oversize_warning/output.8SVX",
          "sha256": "1ef6b90824006a7bb31b91f4c49947419c354f7add0ecc1072d72d58d43d8add"
        },
        "report": {
          "path": "cases/oversize_warning/output_report.json",
          "sha256": "14c60c4160dc4137e3e94969f7256b5f2c46cf730b992a73704c57c101939df8"
        },
        "warnings": [
          "input is 84859 bytes (> 65535). ProTracker may not load the full sample."
        ]
      }
    },
    {
      "id": "silence_minimal",
      "mode": "single",
      "note": "C-2",
      "inputs": [
        {
          "path": "cases/silence_minimal/input.wav",
          "sha256": "150d4cf6b3e2881722fb410df56f7f48da4ec61ab606a487e1e0b14a48239db0"
        }
      ],
      "expect": {
        "output": {
          "path": "cases/silence_minimal/output.8SVX",
          "sha256": "bde940f1f1a59f2b3489f191367a0ac65ec32a58e4bef104fac83a578b0f8867"
        },
        "report": {
          "path": "cases/silence_minimal/output_report.json",
          "sha256": "6605e7beba9d040c6bfd529a3a19ca170a20df6a48d457806ab8e41512073b24"
        }
      }
    }
  ]
}